test/com/facebook/buck/cxx/toolchain/WindowsPreprocessorTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2018-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.cxx.toolchain;
+
+import static org.hamcrest.Matchers.contains;
+import static org.junit.Assert.assertThat;
+
+import com.facebook.buck.rules.Tool;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.easymock.EasyMock;
+import org.hamcrest.junit.ExpectedException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class WindowsPreprocessorTest {
+  @Rule public final ExpectedException expectedException = ExpectedException.none();
+
+  private WindowsPreprocessor windowsPreprocessor;
+
+  @Before
+  public void setUp() {
+    Tool tool = EasyMock.createMock(Tool.class);
+
+    this.windowsPreprocessor = new WindowsPreprocessor(tool);
+  }
+
+  @Test
+  public void testPrefixHeaderArgsHandlesSimpleHeaderFilePaths() {
+    String headerPathStr = "stdafx.h";
+    Path headerPath = Paths.get(headerPathStr);
+
+    assertThat(windowsPreprocessor.prefixHeaderArgs(headerPath), contains("/Yc", headerPathStr));
+  }
+
+  @Test
+  public void testPrefixHeaderArgsHandlesNestedHeaderFilePaths() {
+    String headerPathStr = "include/subdir/stdafx.h";
```

Comment:
Sorry for the back and forth. the windows builds https://ci.appveyor.com/project/Facebook/buck/build/6783 are kinda broken by itself and I missed the error here.

On windows paths are "\" as opposed to "/".
Since this is for Visual C++, which is windows, you should probably make everything "\" and use Junit `Assume(Platform.detect == Platform.WINDOWS)`, with `com.facebook.buck.util.environment.Platform` so that the test only runs on windows.
Make sure to include the corresponding target in the BUCK file.

Reply:
Uhm, sorry but the first symbol in quotes is not shown. I assume you mean `\` ?

Now, we are talking about tests only, right? There is nothing to do on a logic level, since it does not work with any slashes =)

Reply:
Oh, so you're saying tests are broken on a build server? That's why we need this assertion - to prevent running those tests on non-windows OS?

Reply:
The tests currently fail on windows (the appveyor tests) because currently, the test asserts path to be with forward slashes, but on windows will have back slashes. Since we only need this to run on windows properly, we should make the tests assert for backslashes and skip the test for all other platforms.